File: Work_with_SQLite3/src/main/java/Main.java
import java.sql.*;
import java.util.ArrayList;

public class Main {

    public static void main(String[] args) throws SQLException {
        String dbPath = "jdbc:sqlite:C:/Users/A12/Desktop/SQL/Programs/hotel.db";
        Connection connection = DriverManager.getConnection(dbPath);
        Statement statement = connection.createStatement();


        ArrayList<dbRecord> dbRecordArrayList = new ArrayList<>();
        String sqlQuery = "SELECT * FROM guests3";
        ResultSet resultSet = statement.executeQuery(sqlQuery);
        // Из ResultSet данные получаются по одному элементу каждого типа, последовательно
        while(resultSet.next()) {
            dbRecordArrayList.add(new dbRecord(
                    resultSet.getInt("id"),
                    resultSet.getString("name"),
                    resultSet.getString("city"),
                    resultSet.getInt("gender"),
                    resultSet.getInt("age")));
        }
        dbRecordArrayList.forEach(System.out::println);

        connection.close();
    }
}

class dbRecord {
    int id;
    String name;
    String city;
    int gender;
    int age;

    public dbRecord(int id, String name, String city, int gender, int age) {
        this.id = id;
        this.name = name;
        this.city = city;
        this.gender = gender;
        this.age = age;
    }
    @Override
    public String toString() {
        return id + " " + name + " " + city + " " + gender + " " + age;
    }
}

File: Work_with_SQLite3/src/main/java/Main.kt
import java.sql.DriverManager

fun main() {
    val dbPath = "jdbc:sqlite:C:/Users/A12/Desktop/SQL/Programs/hotel.db";
    val connection = DriverManager.getConnection(dbPath)
    val statement = connection.createStatement()
    val resultSet = statement.executeQuery("SELECT * FROM guests3")
    val dbRecordsList = ArrayList<KtDbRecord>()
    while(resultSet.next()) {
        dbRecordsList.add(KtDbRecord(
            resultSet.getInt("id"),
            resultSet.getString("name"),
            resultSet.getString("city"),
            resultSet.getInt("gender"),
            resultSet.getInt("age")));
    }
    dbRecordsList.forEach { println(it) }
}




class KtDbRecord(var id: Int, var name: String, var city: String,
                 var gender: Int, var age: Int) {

    override fun toString(): String {
        return "$id $name $city $gender $age"
    }
}